[Zerolegacy] iP

Zerolegacy opened the pull request:
## Dynamis

> "I dont know what to write here." -myself 

Dynamis frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- **FAST SUPER** FAST to use

All you need to do is,

1. download it from [here](https://github.com/Zerolegacy/ip).
2. Run it.
3. add your tasks.
4. let it manage your tasks for you 😊

And it is **FREE**!

### Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) throws IOException {
        new Dynamis("./data/dynamis.txt").run();
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Renamed the bot and edited welcome message accordingly.

Commit:
Added echo feature

Commit:
Added Todo Class to handle the task features

Commit:
Added Task class and mark as done feature

Commit:
Added Level 3 things and edited previous levels as needed.

Commit:
Added some basic automated tests

Commit:
Added basic error handling

Commit:
Added delete feature

Commit:
Add Save feature to a file on the local system

Commit:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add Save feature to a file on the local system

Commit:
Add DateTime parsing

Commit:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add DateTime parsing

Commit:
Add new classes for more OOP extraction

Commit:
Add files into package

Commit:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

Commit:
Add Junit tests for parser.java and deadline.java

Commit:
Add JavaDocs

Commit:
Edit coding standard

Commit:
Add find feature

Commit:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDocs

Commit:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Edit coding standard

Commit:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add find feature

Commit:
Add gui feature and fix some bugs

Commit:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add gui feature and fix some bugs

lilduckling commented on src/main/java/Dynamis/Dynamis.java:
good job following coding standard!

lilduckling commented on src/main/java/Dynamis/Deadline.java:
better to separate into different folder, i.e: Parser in one folder, Ui in one folder, task-related class in one folder, ect

lilduckling commented on src/main/java/Dynamis/Storage.java:
```diff
@@ -0,0 +1,74 @@
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
```
add comment on what the class is about: java doc

sivaramjeychand commented:
Probably better to split the Dynamis folder into Parser and Tasks

sivaramjeychand commented on src/main/java/Dynamis/Ui.java:
```diff
@@ -0,0 +1,36 @@
+
+import java.util.Scanner;
+
+public class Ui {
+    private final Scanner scanner;
+
```
JavaDocs missing for some Classes and methods such as this one.

sivaramjeychand commented on src/main/java/gui/DialogBox.java:
```diff
@@ -0,0 +1,60 @@
+package gui;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
```
Good job on not using wildcard imports anywhere in your codebase!

sivaramjeychand commented on src/main/java/Dynamis/Task.java:
```diff
@@ -0,0 +1,34 @@
+package Dynamis;
+
+public abstract class Task {
+    protected Boolean doneCheck;
```
Maybe this Boolean could be called isChecked though doneCheck probably is still fine.

Commit:
Add asserts

No asserts are being used.

Assertions help to indicate possible bugs in the code during runtime.

Adding a few assertions in the parseCommand code to help
ensure that input commands are behaving as assumed.

Using assertions here will help detect bugs and exits
which may otherwise be undetected and program continues.

Commit:
Edit code quality

Zerolegacy committed:
Merge pull request #1 from Zerolegacy/branch-A-Assertions

Add asserts

Zerolegacy committed:
Merge branch 'master' into branch-A-CodeQuality

Zerolegacy committed:
Merge pull request #2 from Zerolegacy/branch-A-CodeQuality

Edit code quality

Commit:
Add help feature

Users have no way of viewing a list of available commands
within the chatbot.

Users should be able to access command usage easily without
external documentation.

Add a `help` command that returns a formatted list of all
available commands.

The help command is intuitive to use and is on the main input screen
that users will interact with.

Commit:
Add Ui.png file

Commit:
Edit README.md file to have a better user guide

Commit:
Update user guide and help feature

Commit:
Add javadocs and fix bugs

Commit:
Add more Junit tests
